arm-containerregistry/2017-03-01/swagger/containerregistry.json

```diff
@@ -0,0 +1,613 @@
+        },
+        "reason": {
+          "description": "If any, the reason that the name is not available.",
+          "type": "string"
+        },
+        "message": {
+          "description": "If any, the error message that provides more detail for the reason that the name is not available.",
+          "type": "string"
+        }
+      }
+    },
+    "Registry": {
+      "description": "An object that represents a container registry.",
+      "required": [
+        "sku"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The Sku of the container registry."
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryProperties",
+          "description": "The properties of the container registry.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "Sku": {
+      "description": "The Sku of the container registry.",
+      "properties": {
+        "name": {
+          "description": "Name of the registry Sku. Required for registry creation.",
+          "enum": [
+            "Basic"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": false
+          }
+        },
+        "tier": {
+          "description": "Sku tier based on the Sku name.",
+          "enum": [
+            "Basic"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "SkuTier",
+            "modelAsString": false
+          }
+        }
+      }
+    },
+    "RegistryProperties": {
+      "description": "The properties of a container registry.",
+      "required": [
+        "storageAccount"
+      ],
+      "properties": {
+        "loginServer": {
+          "description": "The URL that can be used to log into the container registry.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationDate": {
+          "format": "date-time",
+          "description": "The creation date of the container registry in ISO8601 format.",
+          "type": "string",
+          "readOnly": true
+        },
+        "adminUserEnabled": {
+          "description": "The value that indicates whether the admin user is enabled. This value is false by default.",
+          "default": false,
+          "type": "boolean"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountProperties",
+          "description": "The properties of the storage account for the container registry. If specified, the storage account must be in the same physical location as the container registry."
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "The properties of a storage account for a container registry.",
+      "required": [
+        "name",
+        "accessKey"
+      ],
+      "properties": {
```

Comment:
Questions regarding storage account
- Is your RP first party resource provider in ARM?
    if Yes, then there is no need to ask for access keys, if not, then make your RP as first party and remove access keys.
- In case you are asking for access keys to be sure that user has access to storage account, then you are doing it wrong way. Rather than asking for Name of storage account, you should ask for ID of storage account and put linked access permission checks.

Let me know if you need more information from me.